x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/grid_embeddable/grid_embeddable.tsx

```diff
@@ -0,0 +1,758 @@
+import { EuiTableActionsColumnType } from '@elastic/eui/src/components/basic_table/table_types';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { Filter } from '@kbn/es-query';
+import { Required } from 'utility-types';
+import {
+  Embeddable,
+  EmbeddableInput,
+  EmbeddableOutput,
+  IContainer,
+} from '../../../../../../../../src/plugins/embeddable/public';
+import { KibanaContextProvider } from '../../../../../../../../src/plugins/kibana_react/public';
+import { DATA_VISUALIZER_GRID_EMBEDDABLE_TYPE } from './constants';
+import { EmbeddableLoading } from './embeddable_loading_fallback';
+import { DataVisualizerStartDependencies } from '../../../../plugin';
+import {
+  IndexPattern,
+  IndexPatternField,
+  KBN_FIELD_TYPES,
+  Query,
+  UI_SETTINGS,
+} from '../../../../../../../../src/plugins/data/common';
+import { SavedSearch } from '../../../../../../../../src/plugins/discover/public';
+import {
+  DataVisualizerTable,
+  ItemIdToExpandedRowMap,
+} from '../../../common/components/stats_table';
+import { FieldVisConfig } from '../../../common/components/stats_table/types';
+import { MetricFieldsStats } from '../../../common/components/stats_table/components/field_count_stats';
+import {
+  getDefaultDataVisualizerListState,
+  getDefaultPageState,
+} from '../../components/index_data_visualizer_view/index_data_visualizer_view';
+
+import { extractSearchData } from '../../utils/saved_search_utils';
+import { useDataVisualizerKibana } from '../../../kibana_context';
+import { extractErrorProperties } from '../../utils/error_utils';
+import { DataLoader } from '../../data_loader/data_loader';
+import { useTimefilter } from '../../hooks/use_time_filter';
+import {
+  DataVisualizerTableState,
+  FieldRequestConfig,
+  JOB_FIELD_TYPES,
+} from '../../../../../common';
+import { kbnTypeToJobType } from '../../../common/util/field_types_utils';
+import { TimeBuckets } from '../../services/time_buckets';
+import { DataVisualizerIndexBasedAppState } from '../../types/index_data_visualizer_state';
+import { IndexBasedDataVisualizerExpandedRow } from '../../../common/components/expanded_row/index_based_expanded_row';
+import { getActions } from '../../../common/components/field_data_row/action_menu';
+import { dataVisualizerRefresh$ } from '../../services/timefilter_refresh_service';
+export type DataVisualizerGridEmbeddableServices = [CoreStart, DataVisualizerStartDependencies];
+export interface DataVisualizerGridEmbeddableInput extends EmbeddableInput {
+  indexPattern: IndexPattern;
+  savedSearch?: SavedSearch;
+  query?: Query;
+  visibleFieldNames?: string[];
+  filters?: Filter[];
+  showPreviewByDefault?: boolean;
+}
+export type DataVisualizerGridEmbeddableOutput = EmbeddableOutput;
+
+export type IDataVisualizerGridEmbeddable = typeof DataVisualizerGridEmbeddable;
+
+const restorableDefaults = getDefaultDataVisualizerListState();
+const defaults = getDefaultPageState();
+
+const useDataVisualizerGridData = (
```

Comment:
Think It might make sense extract this hook to its own file (given the 700 LOC+ )

Reply:
Updated here https://github.com/elastic/kibana/pull/107184/commits/ba1cad7ff6ee7b3391fb42000306c6b3b96096d5